src/components/SignUpCheckEmail.vue

```diff
@@ -0,0 +1,51 @@
+<script setup lang="ts">
+import {
+  getAuth,
+  onAuthStateChanged,
+  sendEmailVerification,
+} from 'firebase/auth';
+import { onBeforeUnmount, ref } from 'vue';
+
+const auth = getAuth();
+
+const email = ref();
+
+async function resendEmail() {
+  try {
+    await sendEmailVerification(auth.currentUser!);
+  } catch (error) {
+    // TODO: Show error in UI.
+    console.log(error);
+  }
+}
+
+const unsubscribeAuthListener = onAuthStateChanged(auth, (user) => {
+  if (user) {
+    email.value = user.email;
+  }
+});
+
+onBeforeUnmount(unsubscribeAuthListener);
+</script>
+
+<template>
+  <h2>Check your email</h2>
+  <p>
+    We've sent an email to {{ email }}. Please click on the link in your email
```

Comment:
nit: this text doesnt match mocks ("We've sent you an email to x.")

Reply:
It doesn't sound grammatically correct to me - I think it's either "We've sent an email to {{ email }}" or "We've sent you an email at {{ email }}".